components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,128 @@
+import styled from "styled-components";
+import { routes } from "../../data/lib";
+import Link from "next/link";
+import { useState } from "react";
+
+export default function SearchBar() {
+  const [filteredData, setFilteredData] = useState([]);
+
+  function handleFilter(event) {
+    const searchEntry = event.target.value;
+    const newFilter = routes.filter((value) => {
+      return value.name.toLowerCase().includes(searchEntry.toLowerCase());
+    });
+    if (searchEntry === "") {
+      setFilteredData([]);
+    } else {
+      setFilteredData(newFilter);
+    }
+  }
+
+  return (
+    <>
+      {filteredData.length != 0 && (
+        <StyledSearch>
+          <StyledSearchResults>
+            {filteredData.slice(0, 10).map((route) => {
```

Comment:
ich versteh nicht genau, warum du slicest...

Reply:
Ich möchte maximal 10 ergebnisse in der Liste anzeigen von allen Übereinstimmungen, deswegen hier der slice auf eine Anzahl von 10 :)